src/main/java/Duke.java

```diff
@@ -1,10 +1,128 @@
+import java.util.Scanner;
+
 public class Duke {
+    private static TaskList taskList = new TaskList();
+    private static Scanner in = initialiseInput();
+    private static Command command;
+    private static String input;
+
+    /**
+     * Main entry point of the application.
+     * Initializes the application and starts the interaction with the user.
+     */
     public static void main(String[] args) {
+        printInitialMsg();
+        do {
+            scanInput();
+            outputReaction();
+        } while (canContinue());
+    }
+
+    private static void printInitialMsg() {
         String logo = " ____        _        \n"
                 + "|  _ \\ _   _| | _____ \n"
                 + "| | | | | | | |/ / _ \\\n"
                 + "| |_| | |_| |   <  __/\n"
                 + "|____/ \\__,_|_|\\_\\___|\n";
         System.out.println("Hello from\n" + logo);
+        System.out.println("Hello! I'm Duke");
+        System.out.println("What can I do for you?");
+    }
+
+    private static void scanInput() {
+        input = takeInput(in);
+        command = processInput(taskList, input);
+    }
+
+    private static void outputReaction() {
```

Comment:
can use verb as the name of the function, "outputReaction" does not sound very clear